File: DoohClick.API/DoohClick.Model/Application/crm/advertiser/AdvertiserModel.cs


namespace DoohClick.Model.Application.crm.advertiser
{
    public record MvAdvertiser
    {
        public int? Id { get; set; }
        public required string Name { get; set; }
        public required string ContactName { get; set; }
        public required string ContactEmail { get; set; }
        public required string ContactPhone { get; set; }
        public required bool? IsActive { get; set; }
        public int? CreatedBy{ get; set; }
        public int? UpdatedBy{ get;set; }
        public string? Creator { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }

    public record MvAdvertiserDdl
    {
        public int? Id { get; set; }
        public required string Name { get; set; }
    }

    public record MvTenantIdParam
    {
        public int? TenantId { get; set; }
    }

    public record MvAdvertiserFilterOptions
    {
        public int? TenantId { get; set; }
        public List<bool>? IsActiveList { get; set; }
    }
}

File: DoohClick.Web/src/app/crm/advertiser/model/advertiser.model.ts
export interface MvAdvertiser {
  id: number;
  name: string;
  contactName: string;
  contactEmail: string;
  contactPhone: string;
  isActive: boolean;
  createdBy: number;
  updatedBy: number | null;
  creator: string | null;
  createdAt: string;
  updatedAt: string | null;
}

export interface MvAdvertiserDdl {
  id: number;
  name: string;
}

export interface MvTenantIdParam {
  tenantId: number;
}

export interface MvAdvertiserFilterOptions {
  tenantId: number;
  isActiveList: boolean[] | null;
}
